File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> f;
        for(int x: nums) f[x]++;
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<>> p;
        for(auto& q: f) {
            p.push({q.second, q.first});
            if(p.size() > k) p.pop();
        }
        vector<int> ans;
        while(!p.empty()) {
            ans.push_back(p.top().second);
            p.pop();
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        ans = 0
        l = 0
        n = len(s)
        mp = {}
        m = 0
        for r in range(n):
            mp[s[r]] = mp.get(s[r], 0)+1
            m = max(m, mp[s[r]])
            while m < r-l-k+1:
                mp[s[l]] -= 1
                l += 1
            ans = max(ans, r-l+1)
        return ans